Lib/test/datetimetester.py

```diff
@@ -61,7 +61,7 @@ def test_constants(self):
         self.assertEqual(datetime.MAXYEAR, 9999)
 
     def test_name_cleanup(self):
-        if '_Fast' not in str(self):
+        if self._test_type == 'Pure':
```

Comment:
I'd rather have one property serve one purpose: `__qualname__` to show correct class names in verbose mode (cosmetic) and `_test_type` for determining what kind of test it is (functional). This would, hopefully, prevent future cosmetic changes from interfering with the functional parts.

However, if you think `"_Fast" in self.__class__.__qualname__` will minimize changes, I'll be happy to change it.

PS: I believe at some point in time, `unittest` used the `cls.__name__` in verbose mode instead of `cls.__qualname__`. This would explain the change in the outputs which surprised @haypo. It would be ideal if such cosmetic changes do not break other parts of the code.